pkg/agent/validations_test.go

```diff
@@ -2,114 +2,271 @@ package agent
 
 import (
 	"io/ioutil"
-	"regexp"
 	"testing"
 
-	"github.com/openshift/assisted-service/models"
 	"github.com/sirupsen/logrus"
-	"github.com/sirupsen/logrus/hooks/test"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestCheckHostsValidation(t *testing.T) {
+// Test logger
+var logger = &logrus.Logger{
+	Out:       ioutil.Discard,
+	Formatter: new(logrus.TextFormatter),
+	Hooks:     make(logrus.LevelHooks),
+	Level:     logrus.DebugLevel,
+}
+
+// ValidationHistory test helpers
+const (
+	testKey                string = "test-cluster"
+	testID                 string = "test"
+	testLogPrefix          string = "Test: "
+	validationsInfoSuccess string = "{\"test\":[{\"id\":\"test\",\"status\":\"success\",\"message\":\"The validation succeeded\"}]}"
+	validationsInfoFailure string = "{\"test\":[{\"id\":\"test\",\"status\":\"failure\",\"message\":\"The validation failed\"}]}"
+)
+
+func TestUpdateValidationHistory(t *testing.T) {
 	tests := []struct {
-		name           string
-		hosts          []*models.Host
-		expectedResult bool
-		expectedLogs   []string
```

Comment:
Would be possible to keep the `expectedLogs` field (if not here, at leasr in another test)? Its main intent was to capture the output seen by the user under different conditions, and I think it would be useful to keep it (also as a form of internal dev documentation)

Reply:
This logic actually wasn't checking for the output of the log clearly. It was just checking if any output log output on the desired log level. I spent a couple of hours fixing this test when I switched to using Debug over Error and it wasn't clear as to why. I disagree that it was of any use.

Reply:
It looks to me like the code on lines 101-112 was checking that there was at least one log message matching each of the `expectedLogs` (as a regex)?

Reply:
It matches the regex thats true but what is the value of testing the matching regex? Are we writing a unit test for regex or our code?

Reply:
Ideally when we're checking the level we should also test that we're checking the level _of the right message_, so that we can't accidentally lose test coverage by adding another log. Probably no need to have an `expectedLogs` field though, since we know it should always report the message.

Reply:
I did some refactoring on `TestLogValidationResultHistory()` that should address all these comments.